File: android/src/main/java/expo/modules/gaodemap/overlays/MarkerViewModule.kt
package expo.modules.gaodemap.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Marker 视图 Module
 */
class MarkerViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("MarkerView")

    View(MarkerView::class) {
      Events("onPress", "onDragStart", "onDrag", "onDragEnd")
      
      // 拆分 position 为两个独立属性以兼容 React Native 旧架构
      Prop<Double>("latitude") { view, lat ->
        view.setLatitude(lat)
      }
      
      Prop<Double>("longitude") { view, lng ->
        view.setLongitude(lng)
      }
      
      Prop<String>("title") { view, title ->
        view.setTitle(title)
      }
      
      Prop<String>("description") { view, description ->
        view.setDescription(description)
      }
      
      Prop<Boolean>("draggable") { view, draggable ->
        view.setDraggable(draggable)
      }
      
      Prop<Float>("opacity") { view, opacity ->
        view.setOpacity(opacity)
      }
      
      Prop<Boolean>("flat") { view, flat ->
        view.setFlat(flat)
      }
      
      Prop<Float>("zIndex") { view, zIndex ->
        view.setZIndex(zIndex)
      }
      
      Prop<Map<String, Float>>("anchor") { view, anchor ->
        view.setAnchor(anchor)
      }
      
      Prop<Int>("iconWidth") { view, width ->
        view.setIconWidth(width)
      }
      
      Prop<Int>("iconHeight") { view, height ->
        view.setIconHeight(height)
      }
      
      Prop<Int>("customViewWidth") { view, width ->
        view.setCustomViewWidth(width)
      }
      
      Prop<Int>("customViewHeight") { view, height ->
        view.setCustomViewHeight(height)
      }
    }
  }
}

File: ios/overlays/MarkerViewModule.swift
import ExpoModulesCore

public class MarkerViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("MarkerView")
        
        View(MarkerView.self) {
            // 拆分 position 为两个独立属性以兼容 React Native 旧架构
            Prop("latitude") { (view: MarkerView, lat: Double) in
                view.setLatitude(lat)
            }
            
            Prop("longitude") { (view: MarkerView, lng: Double) in
                view.setLongitude(lng)
            }
            
            Prop("title") { (view: MarkerView, title: String) in
                view.setTitle(title)
            }
            
            Prop("snippet") { (view: MarkerView, snippet: String) in
                view.setDescription(snippet)
            }
            
            Prop("draggable") { (view: MarkerView, draggable: Bool) in
                view.setDraggable(draggable)
            }
            
            Prop("icon") { (view: MarkerView, source: [String: Any]?) in
                view.setIcon(source)
            }
            
            Prop("iconWidth") { (view: MarkerView, width: Double) in
                view.iconWidth = width
            }
            
            Prop("iconHeight") { (view: MarkerView, height: Double) in
                view.iconHeight = height
            }
            
            Prop("customViewWidth") { (view: MarkerView, width: Double) in
                view.customViewWidth = width
            }
            
            Prop("customViewHeight") { (view: MarkerView, height: Double) in
                view.customViewHeight = height
            }
            
            Prop("centerOffset") { (view: MarkerView, offset: [String: Double]) in
                view.setCenterOffset(offset)
            }
            
            Prop("animatesDrop") { (view: MarkerView, animate: Bool) in
                view.setAnimatesDrop(animate)
            }
            
            Prop("pinColor") { (view: MarkerView, color: String) in
                view.setPinColor(color)
            }
            
            Prop("canShowCallout") { (view: MarkerView, show: Bool) in
                view.setCanShowCallout(show)
            }
        }
    }
}